frontend/src/components/module-viewer/content/modals/EditFlipCardModal.tsx

```diff
@@ -28,9 +28,9 @@ const EditFlipCardModal = ({
   const [tabIndex, setTabIndex] = useState(0);
 
   const addCard = () => {
-    setTabIndex(cards.length);
-    setCards([
-      ...cards,
+    setTabIndex(() => cards.length);

```

Comment:
Looks mostly good! I would move this inside the below callback to pass in state.length instead of cards.length (since the dependency is on cards, not the existing tabIndex). Then this setState doesn't have to be functional.

Reply:
Are you saying I should put `setTabIndex` inside the `setCards` function?